estimate_ground_surface:
  ros__parameters:
    rotation_angle:
      # thetaX: 3.0
      # thetaY: 12.0
      # thetaZ: 0.0
      roll_difference: 0.00
      pitch_difference: 0.00

    lidar_height:
      # z_offset: 1.15
      z_offset: 1.3

    crop_box:
      min:
        x: 0.5
        y: -5.0
        z: -3.0
        a: 1.0
      max:
        x: 10.0
        y: 5.0
        z: 3.0
        a: 1.0

    grid:
      min_x: -10.0
      max_x: 10.0
      min_y: -10.0
      max_y: 10.0
      resolution: 0.5
      variance: 0.05
      cluster_size: 10
      z_threshold_low: -0.4
      z_threshold_high: 0.4
      ratio_threshold: 0.5
    
    closest_points:
      search_min_x: 1.0
      search_max_x: 4.0
      search_min_y: -3.0
      search_max_y: 3.0
    
    x_division:
      boundary: 3.0       # 境界のX座標（m）
      min_points: 100     # 各セクションでの最低点数
      plane_angle_threshold: 20.0      # 平面間角度閾値（度）
      plane_distance_threshold: 0.1